### PR TITLE
#44 Implement profile content

### LINKED ISSUE #44: Implement Profile content

### PR DESCRIPTION
**Pull request by devraulu**
Fixes #44

### CODE REVIEW

File: .eslintcache [hunk 1]
```diff
@@ -0,0 +1 @@
...nents\\Reviews.jsx",["132","133"],"D:\\Web Dev\\Mentorpaired\\frontend_staging\\frontend\\src\\modules\\profile\\components\\Goals.jsx",[],"D:\\Web Dev\\Mentorpaired\\frontend_staging\\frontend\\src\\modules\\profile\\components\\RatingComments.jsx",[],"D:\\Web Dev\\Mentorpaired\\frontend_staging\\frontend\\src\\modules\\profile\\components\\DropdownButton.jsx",[],"D:\\Web Dev\\Mentorpaired\\frontend_staging\\frontend\\src\\modules\\profile\\components\\Dropdown.jsx",[],{"ruleId":"134","severity":1,"message":"135","line":6,"column":8,"nodeType":"136","messageId":"137","endLine":6,"endColumn":15},{"ruleId":"134","severity":1,"message":"138","line":2,"column":8,"nodeType":"136","messageId":"137","endLine":2,"endColumn":20},{"ruleId":"134","severity":1,"message":"139","line":3,"column":8,"nodeType":"136","messageId":"137","endLine":3,"endColumn":18},{"ruleId":"134","severity":1,"message":"140","line":4,"column":8,"nodeType":"136","messageId":"137","endLine":4,"endColumn":22},"no-unused-vars","'Profile' is defined but never used.","Identifier","unusedVar","'MentorButton' is defined but never used.","'filterIcon' is defined but never used.","'DropdownButton' is defined but never used."]
```
Comment (toystars) [inline .eslintcache]: I don't seem to understand the purpose of this file 🤔

File: package.json [hunk 2]
```diff
@@ -3,20 +3,21 @@
...
   "dependencies": {
-    "@fortawesome/fontawesome-free": "^5.15.1",
     "@fortawesome/fontawesome-svg-core": "^1.2.32",
     "@fortawesome/free-brands-svg-icons": "^5.15.1",
     "@fortawesome/free-solid-svg-icons": "^5.15.1",
     "@fortawesome/react-fontawesome": "^0.1.13",
-    "@testing-library/jest-dom": "^5.11.5",
-    "@testing-library/react": "^11.1.2",
-    "@testing-library/user-event": "^12.2.2",
+    "@testing-library/jest-dom": "^5.11.6",
```
Comment (toystars) [inline package.json]: test packages should be installed as `devDependencies`.

File: src/App.js [hunk 3]
```diff
@@ -1,21 +1,92 @@
...
-          <Switch>
-            <Route exact path="/" component={DashboardPage} />
-            <Route path="/profile" component={DashboardPage} />
-            <Route path="/messages" component={DashboardPage} />
-            <Route path="/signin" component={SignIn} />
-          </Switch>
-      </BrowserRouter>
-    );
-  }
+export const UserContext = React.createContext();
```
Comment (toystars) [inline src/App.js]: The context object, initialState and reducer could be moved to a separate file. Maybe a `store.js` file? The App.js file becomes unnecessarily long with this.

File: src/index.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+.main-wrapper {
+  display: flex;
+  overflow: hidden;
+}
+
+.main-wrapper .page-wrapper {
+  min-height: 100vh;
+  background: #f7f8fa;
+  width: calc(100% - 280px);
```
Comment (toystars) [inline src/index.scss]: I am not comfortable with this calculation here. What happens on mobile where the device width is less than 320px? How many widths would be assigned?

File: src/modules/profile/components/profile.styles.scss [hunk 5]
```diff
@@ -0,0 +1,285 @@
+// Variables
```
Comment (toystars) [inline src/modules/profile/components/profile.styles.scss]: I like the use of sass here; variables and all. Do you think this stylesheet could be made simpler? I think it is overcomplicated. Also, having all the styles of a page in a single CSS file tends to be cumbersome and difficult to maintain. I would propose every component to have its own stylesheet imported to the CSS files would be kept as short and readable as possible.

**Review by toystars — CHANGES_REQUESTED** [review]
Please make use of `rem` as a spacing unit. You can use [this tool](https://nekocalc.com/px-to-rem-converter) for conversions.

File: .eslintcache (hunk 1 above)
Comment (devraulu) [reply .eslintcache]: I think it should be added to `.gitignore`.

File: package.json (hunk 2 above)
Comment (devraulu) [reply package.json]: I'm pretty sure that's automatically installed by `create-react-app`, what do you suggest I do about it?

File: src/index.scss (hunk 4 above)
Comment (devraulu) [reply src/index.scss]: I didn't do this, I think is from the old design. I will change it to full width.

File: src/modules/profile/components/profile.styles.scss (hunk 5 above)
Comment (devraulu) [reply src/modules/profile/components/profile.styles.scss]: Done. If you have any other suggestions on this topic please let me know.

File: package.json (hunk 2 above)
Comment (toystars) [reply package.json]: @devCayacoa you can move them to dev dependencies

**Review by toystars — APPROVED** [review]

**Pull request merged**